File: backend/app/schemas/product.py
from sqlalchemy import Column, String, Text
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class Product(Base):
    __tablename__ = "product"
    id = Column(String(100), primary_key=True, unique=True, nullable=False)
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    image = Column(String(255), nullable=True)
    category = Column(String(100), nullable=False)
    subCategory = Column(String(100), nullable=False)
    slug = Column(String(255), nullable=True)
    features = Column(Text, nullable=True)            # JSON.stringify
    datasheet = Column(String(512), nullable=True)
    detail_key_features = Column(Text, nullable=True) # JSON array
    detail_images = Column(Text, nullable=True)       # JSON array

File: fe/src/types/product.ts
export interface ApiProductFeature {
  title: string;
  subFeatures: string[];
}

export interface ApiProduct {
  id: string;
  name: string;
  description?: string;
  image?: string;
  category: string;
  subCategory: string;
  slug?: string;
  features?: ApiProductFeature[];
  datasheet?: string;
  detail_key_features?: string[];
  detail_images?: string[];
}

export interface UiProduct {
  id: string;
  name: string;
  description: string;
  image: string;
  category: string;
  subCategory: string;
  slug?: string;
  features?: ApiProductFeature[];
  datasheet?: string;
}

export interface UiProductDetail {
  id: string;
  slug: string;
  name: string;
  images: string[];
  overview: {
    summary: string;
    keyFeatures: string[];
    datasheetLink: string;
  };
  features?: ApiProductFeature[];
  datasheet?: string;
}


